google/cloud/bigtable/table_admin_legacy_test.cc

```diff
@@ -95,159 +95,190 @@ class TableAdminTest : public ::testing::Test {
   }
 
-  std::shared_ptr<MockAdminClient> client_ =
-      std::make_shared<MockAdminClient>();
-};
+  Status IsContextMDValid(grpc::ClientContext& context,
+                          std::string const& method) {
+    return validate_metadata_fixture_.IsContextMDValid(
+        context, method, google::cloud::internal::ApiClientHeader());
+  }
 
-// A lambda to create lambdas.  Basically we would be rewriting the same
-// lambda twice without this thing.
-auto create_list_tables_lambda =
-    [](std::string const& expected_token, std::string const& returned_token,
-       std::vector<std::string> const& table_names) {
-      return [expected_token, returned_token, table_names](
-                 grpc::ClientContext* context,
-                 btadmin::ListTablesRequest const& request,
-                 btadmin::ListTablesResponse* response) {
-        EXPECT_STATUS_OK(IsContextMDValid(
-            *context, "google.bigtable.admin.v2.BigtableTableAdmin.ListTables",
-            google::cloud::internal::ApiClientHeader()));
-        auto const instance_name =
-            "projects/" + kProjectId + "/instances/" + kInstanceId;
-        EXPECT_EQ(instance_name, request.parent());
-        EXPECT_EQ(btadmin::Table::FULL, request.view());
-        EXPECT_EQ(expected_token, request.page_token());
+  Status IsContextMDValid(grpc::ClientContext& context,
+                          std::string const& method,
+                          std::string const& resource_name) {
+    return validate_metadata_fixture_.IsContextMDValid(
+        context, method, google::cloud::internal::ApiClientHeader(),
+        resource_name);
+  }
 
-        EXPECT_NE(nullptr, response);
-        for (auto const& table_name : table_names) {
-          auto& table = *response->add_tables();
-          // NOLINTNEXTLINE(performance-inefficient-string-concatenation)
-          table.set_name(instance_name + "/tables/" + table_name);
-          table.set_granularity(btadmin::Table::MILLIS);
-        }
-        // Return the right token.
-        response->set_next_page_token(returned_token);
-        return grpc::Status::OK;
-      };
+  std::function<grpc::Status(grpc::ClientContext* context,
+                             btadmin::ListTablesRequest const& request,
+                             btadmin::ListTablesResponse* response)>
+  CreateListTablesMock(std::string const& expected_token,
+                       std::string const& returned_token,
+                       std::vector<std::string> const& table_names) {
+    return [this, expected_token, returned_token, table_names](
+               grpc::ClientContext* context,
+               btadmin::ListTablesRequest const& request,
+               btadmin::ListTablesResponse* response) {
+      EXPECT_STATUS_OK(IsContextMDValid(
+          *context, "google.bigtable.admin.v2.BigtableTableAdmin.ListTables",
+          google::cloud::internal::ApiClientHeader()));
```

Comment:
This is setting the `resource_name` to `google::cloud::internal::ApiClientHeader()`... I'm guessing you don't want that.

Reply:
Fixed.  I think that says something about our tests.